src/plugins/intel_cpu/src/nodes/rdft.h

```diff
@@ -94,6 +95,66 @@ struct RDFTExecutor {
     std::vector<float> generateTwiddlesCommon(size_t inputSize, size_t outputSize, enum dft_type type, bool useFFT);
 };
 
+struct RDFTRefExecutor : public RDFTExecutor {
```

Comment:
Can we return `RDFTRefExecutor` and `RDFTJitExecutor` definition to the source file and just create static builder-method in the base class: 
```
std::shared_ptr<RDFTExecutor> build(bool inverse, NodeDesc* primDesc = nullptr) {
#if defined(OPENVINO_ARCH_X86_64)
        if (mayiuse(cpu::x64::sse41)) {
            return std::make_shared<RDFTJitExecutor>(key.isInverse, primDesc);
        }
#endif
        const auto executor = std::make_shared<RDFTRefExecutor>(key.isInverse);
        primDesc->setImplementationType(ref_any);
        return executor;
}
```
This will help us hide all the logic related to RDF from the eyes of developers 😊 Then in `STFT` you need to just call `RDFTExecutor::make(...)` in `PrepareParams`. What do you think?

Reply:
Good point, it requires less changes and less code in rdft.h header file. Thanks!
The only moved struct is the RDFTKey.
The `RDFTExecutor::build` is actually called in `STFT::createPrimitive()` instead of `PrepareParams` following the existing `RDFT::createPrimitive`.